backend/src/main/java/com/soen341/instagram/security/AuthorizationServerConfig.java

```diff
@@ -0,0 +1,45 @@
+package com.soen341.instagram.security;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.AuthenticationManager;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.security.oauth2.config.annotation.configurers.ClientDetailsServiceConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configuration.AuthorizationServerConfigurerAdapter;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableAuthorizationServer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerEndpointsConfigurer;
+import org.springframework.security.oauth2.config.annotation.web.configurers.AuthorizationServerSecurityConfigurer;
+
+@Configuration
+@EnableAuthorizationServer
+public class AuthorizationServerConfig extends AuthorizationServerConfigurerAdapter
+{
+	@Autowired
+	private AuthenticationManager authenticationManager;
+
+	@Autowired
+	private PasswordEncoder passwordEncoder;
+
+	@Override
+	public void configure(final AuthorizationServerSecurityConfigurer security) throws Exception
+	{
+		// returns true if user not anonymous
+		security.checkTokenAccess("isAuthenticated()");
+	}
+
+	@Override
+	public void configure(final ClientDetailsServiceConfigurer clients) throws Exception
+	{
+		clients.inMemory().withClient("instagram_client").authorizedGrantTypes("client_credentials", "password")
```

Comment:
Need to add grant type "refresh_token" so that an access token request returns the refresh token as well